src/components/RecipientItem.vue

```diff
@@ -24,18 +29,48 @@
 <script lang="ts">
   import { Options, Vue } from "vue-class-component"
 
+  import { showSpinnerMethod, replaceWithLoader } from "@/utils/showSpinner"
+  import { debounceMethod } from "@/utils/debounce"
+  import applyDecorators from "@/utils/applyDecorators"
+  import { UIError } from "@/exception"
+
   import moment from "moment"
 
   @Options({
     name: "RecipientItem",
     props: {
       recipient: Object,
     },
+    methods: {
+      toggleFavorite: applyDecorators(
+        [
+          debounceMethod,
+          showSpinnerMethod(function (this: any) {
+            return replaceWithLoader.apply(this, ["#toggle-fav", "1.5em"])
```

Comment:
doesn't seem to do what was expected on my end : I can see both the favorite icon (the star) and the loading symbol. And they are misaligned.